backend/controllers/calculation_controller.go

```diff
@@ -0,0 +1,143 @@
+package controllers
+
+import (
+	"fmt"
+	"mmr/backend/mmr"
+	view "mmr/backend/models"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/intinig/go-openskill/rating"
+	"github.com/intinig/go-openskill/types"
+)
+
+type CalculationController struct{}
+
+// SubmitMMRCalculation godoc
+//
+//	@Summary		Submit an MMR calculation request
+//	@Description	Submit two teams' details for MMR calculation
+//	@Tags 			MMR
+//	@Accept			json
+//	@Produce		json
+//	@Param			request	body		view.MMRCalculationRequest	true	"MMR Calculation Request"
+//	@Success		200		{object}	view.MMRCalculationResponse	"MMR calculation result"
+//	@Router			/v2/mmr/calculate [post]
+func (m CalculationController) SubmitMMRCalculation(c *gin.Context) {
```

Comment:
This needs to be setup in router.go
Also needs to rely on api key and not jwt like the rest of the mmr endpoints